code/modules/mob/inventory.dm

```diff
@@ -1,3 +1,7 @@
+//This variable is needed in order to remember the origin of the storage of an item.
+// GLOBAL_VAR_INIT(quick_equip_memory_item, 0)
+// GLOBAL_VAR_INIT(quick_equip_memory_origin, 0)
+GLOBAL_VAR_INIT(quick_equip_cowboy_delay_negation, 0)
```

Comment:
This should be a var on guns themselves, especially since we have access to the specific gun being equipped. Why? Global variables are generally used for things that need to have a certain state shared across multiple other things, like cached data, shared cooldowns, and such.

Reply:
oh yeah, I don't know why I didn't do it like that to begin with. Allow me to change that

Reply:
See Iteration VII